tests/test_utils.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import numpy as np
+
+from cata2data import collate_fn_irregular_cutouts, collate_fn_regular_cutouts
+
+def test_collate_fn_irregular_cutouts_with_arraylike_y():
+    batch = [
+        (np.random.rand(3, 4), np.array([1, 2, 3])),
+        (np.random.rand(3, 4), np.array([4, 5, 6])),
+    ]
+    x, y = collate_fn_irregular_cutouts(batch)
+
+    assert isinstance(x, list), "Expected x to be a list."
+    assert isinstance(y, torch.Tensor), "Expected y to be a tensor."
+    assert len(x) == 2, "Expected x to have length 2."
+    assert y.shape == (6,), "Expected y to be concatenated correctly."
```

Comment:
Should this be `6`? Should it not be `[2,3]`?

Reply:
Since the input array is an `np.ndarray` the data is concatenated into a 1d array by this line of the function 
```python
if isinstance(y[0], np.ndarray) or isinstance(y[0], list):
        y = torch.as_tensor(np.concatenate([target.astype(np.float32) for target in y]))
```
Is this not what is intended?

Reply:
I don't think that is what is intended. I think I probably should have used vstack:
```python
if isinstance(y[0], np.ndarray) or isinstance(y[0], list):
        y = torch.as_tensor(np.vstack([target.astype(np.float32) for target in y]))
```
Otherwise for multitarget problems (i.e. multiclass classification or multivariate regression) the `N` targets would be in a 1D array of length `batch_size*N` which is much less practical than 2D, ie. `[batch_size,N]`.

Reply:
Makes sense. Have pushed that change and updated the test